x-pack/plugins/stack_connectors/server/connector_types/gen_ai/gen_ai.ts

```diff
@@ -157,4 +182,22 @@ export class GenAiConnector extends SubActionConnector<GenAiConfig, GenAiSecrets
 
     return { available: response.success };
   }
+
+  /**
+   * takes an array of messages and a model as input and returns a promise that resolves to a string.
+   * Sends the stringified input to the runApi method. Returns the trimmed completion from the response.
+   * @param body An object containing array of message objects, and possible other OpenAI properties
+   */
+  public async invokeAI(body: InvokeAIActionParams): Promise<InvokeAIActionResponse> {
+    const res = await this.runApi({ body: JSON.stringify(body) });
+
+    if (res.choices && res.choices.length > 0 && res.choices[0].message.content) {
```

Comment:
nit: use `res.choices[0].message?.content` in case of malformed response messages.